lib/publish.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+const execa = require('execa');
+const listrInput = require('listr-input');
+const Observable = require('any-observable');
+
+const npmPublish = opts => {
+	const args = ['publish'];
+
+	if (opts.tag) {
+		args.push('--tag', opts.tag);
+	}
+
+	if (opts.otp) {
+		args.push('--otp', opts.otp);
+	}
+
+	return execa('npm', args);
+};
+
+module.exports = tag => Observable.fromPromise(npmPublish({tag}))
+	.catch(err => {
+		if (err.stderr.indexOf('You must provide a one-time pass') !== -1) {
+			return listrInput('Enter OTP', {
+				done: otp => npmPublish({tag, otp})
```

Comment:
In the case of an invalid otp, should this `npmPublish` throw?

Reply:
Good question, maybe just ask it again. I'll try it out.